tests/test_directives.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import sys
+from copy import deepcopy
+
+from flow.directives import Directives
+from flow.directives import NP, NRANKS, NGPU, EXECUTABLE, OMP_NUM_THREADS
+
+
+@pytest.fixture()
+def setUp(test_class=Directives, **kwargs):
+    def _setup(test_class=Directives, **kwargs):
+        return test_class(**kwargs)
+    return _setup
+
+
+available_directives_list = [NP, NRANKS, NGPU, EXECUTABLE, OMP_NUM_THREADS]
+
+
+class TestItems:
+    """
+    Tests for _DirectivesItem class
+    """
+    def test_serial(self):
+        assert NP.serial(4, 2) == 4
+        assert not NP.serial('4', '2') == 4
+
+    def test_parallel(self):
+        assert NP.parallel(4, 2) == 6
+
+    def test_parallel_manual(self):
+        directive_copy = deepcopy(NP)
+        directive_copy.parallel = lambda x, y: x*y/(x+y)
+        assert directive_copy.parallel(3, 2) == 1.2
+        with pytest.raises(TypeError):
+            directive_copy.parallel('4', '2')
+        with pytest.raises(ZeroDivisionError):
+            directive_copy.parallel(0, 0)
+
+    def test_validation(self):
+        for directive in available_directives_list:
+            directive.validation(directive.default)
+
+    def test_validation_manual(self):
+        NP_copy = deepcopy(NP)
+        NP_copy.default = 0
```

Comment:
Why do we need to set the default to 0 we can just directly test the validation.

Reply:
Code has not change. Is not resolved.